src/SFA.DAS.PR.Jobs/MessageHandlers/EmployerAccounts/AddedLegalEntityEventHandler.cs

```diff
@@ -1,47 +1,80 @@
+    IEmployerAccountsApiClient _employerAccountsApiClient 
+) : IHandleMessages<AddedLegalEntityEvent>
 {
     public const string AccountLegalEntityAlreadyExistsFailureReason = "Account legal entity already exists";
 
     public async Task Handle(AddedLegalEntityEvent message, IMessageHandlerContext context)
     {
-        var accountLegalEntity = await _providerRelationshipsDataContext
+        AccountLegalEntity? accountLegalEntity = await _providerRelationshipsDataContext
             .AccountLegalEntities
             .FirstOrDefaultAsync(a => a.AccountId == message.AccountId && a.Id == message.AccountLegalEntityId, context.CancellationToken);
 
         if (accountLegalEntity != null)
         {
             _logger.LogWarning("Legal entity with Id:{LegalEntityId} already exists", message.LegalEntityId);
+
             JobAudit jobAudit = new(
-                nameof(AddedLegalEntityEventHandler),
-                new EventHandlerJobInfo<AddedLegalEntityEvent>(context.MessageId, message, false, AccountLegalEntityAlreadyExistsFailureReason));
+                nameof(AddedLegalEntityEventHandler), 
+                new EventHandlerJobInfo<AddedLegalEntityEvent>(context.MessageId, message, false, AccountLegalEntityAlreadyExistsFailureReason)
+            );
+
             _providerRelationshipsDataContext.JobAudits.Add(jobAudit);
         }
         else
         {
-            _providerRelationshipsDataContext
-                .AccountLegalEntities
-                .Add(new()
+            Account? providerRelationshipsAccount = await _providerRelationshipsDataContext.Accounts.FirstOrDefaultAsync(a => a.Id == message.AccountId, context.CancellationToken);
+
+            if(providerRelationshipsAccount is null)
+            {
+                var accountResponse = await _employerAccountsApiClient.GetAccount(message.AccountId, context.CancellationToken);
+
+                providerRelationshipsAccount = new Account()
                 {
-                    Id = message.AccountLegalEntityId,
-                    AccountId = message.AccountId,
-                    PublicHashedId = message.AccountLegalEntityPublicHashedId,
-                    Name = message.OrganisationName,
-                    Created = message.Created
-                });
-            _providerRelationshipsDataContext
-                .JobAudits
-                .Add(new(nameof(AddedLegalEntityEventHandler),
-                    new EventHandlerJobInfo<AddedLegalEntityEvent>(context.MessageId, message, true, null)));
+                    Id = message.AccountId,
+                    HashedId = accountResponse.HashedAccountId,
+                    PublicHashedId = accountResponse.PublicHashedAccountId,
+                    Name = accountResponse.DasAccountName,
+                    Created = DateTime.UtcNow
+                };
+
+                await _providerRelationshipsDataContext.Accounts.AddAsync(providerRelationshipsAccount, context.CancellationToken);
```

Comment:
Do not use AddAsync

Reply:
This is actually not required as we are adding the entity object to ALE entity so it will get added if it is new.